ext/libdatadog_api/feature_flags.c

```diff
@@ -0,0 +1,430 @@
+static VALUE resolution_details_get_variant(VALUE self);
+static VALUE resolution_details_get_allocation_key(VALUE self);
+static VALUE resolution_details_get_do_log(VALUE self);
+
+static VALUE native_get_assignment(VALUE self, VALUE config, VALUE flag_key, VALUE context);
+
+
+void feature_flags_init(VALUE open_feature_module) {
+  // Always define the Binding module - it will reuse existing if it exists
+  VALUE binding_module = rb_define_module_under(open_feature_module, "Binding");
+
+  // Configuration class
+  VALUE configuration_class = rb_define_class_under(binding_module, "Configuration", rb_cObject);
+  rb_define_alloc_func(configuration_class, configuration_alloc);
+  rb_define_method(configuration_class, "_native_initialize", configuration_initialize, 1);
+
+  // EvaluationContext class  
+  VALUE evaluation_context_class = rb_define_class_under(binding_module, "EvaluationContext", rb_cObject);
+  rb_define_alloc_func(evaluation_context_class, evaluation_context_alloc);
+  rb_define_method(evaluation_context_class, "_native_initialize_with_attributes", evaluation_context_initialize_with_attributes, 2);
+
+  // ResolutionDetails class
+  VALUE resolution_details_class = rb_define_class_under(binding_module, "ResolutionDetails", rb_cObject);
+  rb_define_alloc_func(resolution_details_class, resolution_details_alloc);
+
+  rb_define_method(resolution_details_class, "value", resolution_details_get_value, 0);
+  rb_define_method(resolution_details_class, "reason", resolution_details_get_reason, 0);
+  rb_define_method(resolution_details_class, "error_code", resolution_details_get_error_code, 0);
+  rb_define_method(resolution_details_class, "error_message", resolution_details_get_error_message, 0);
+  rb_define_method(resolution_details_class, "variant", resolution_details_get_variant, 0);
+  rb_define_method(resolution_details_class, "allocation_key", resolution_details_get_allocation_key, 0);
+  rb_define_method(resolution_details_class, "do_log", resolution_details_get_do_log, 0);
+
+  // Module-level method
+  rb_define_module_function(binding_module, "_native_get_assignment", native_get_assignment, 3);
+}
+
+// Configuration TypedData definition
+static const rb_data_type_t configuration_typed_data = {
+  .wrap_struct_name = "Datadog::OpenFeature::Binding::Configuration",
+  .function = {
+    .dmark = NULL,
+    .dfree = configuration_free,
+    .dsize = NULL,
+  },
+  .flags = RUBY_TYPED_FREE_IMMEDIATELY
+};
+
+static VALUE configuration_alloc(VALUE klass) {
+  ddog_ffe_Handle_Configuration *config = ruby_xcalloc(1, sizeof(ddog_ffe_Handle_Configuration));
+  *config = NULL; // Initialize the handle to NULL
+  return TypedData_Wrap_Struct(klass, &configuration_typed_data, config);
+}
+
+static void configuration_free(void *ptr) {
+  ddog_ffe_Handle_Configuration *config = (ddog_ffe_Handle_Configuration *) ptr;
+  if (config && *config) {
+    ddog_ffe_configuration_drop(config);
+  }
+  ruby_xfree(ptr);
+}
```

Comment:
nit: not entirely sure about this but handle *is* a pointer and it fits to data field (`void *`), so I think we can avoid allocations here, call `rb_undef_alloc_func(configuration_class)`, and call `TypedData_Wrap_Struct` in `configuration_initialize`.

```suggestion
static void configuration_free(void *ptr) {
  ddog_ffe_configuration_drop(&(ddog_ffe_Handle_Configuration) ptr);
}
```

Similarly, for all other handle types